### ISSUE TITLE
how about adding component="file" to Field?

### ISSUE THREAD

**Issue by seport** [erikras/redux-form#3993]
### Are you submitting a **bug report** or a **feature request**?

Feature Request

### Other information

Currently, it seems that setting up a file input in redux form requires passing a custom field to a `Field` component and wiring it up with event listeners.
Since file inputs are pretty common, I was thinking it would be nice to expand the `component` prop on `Field` which already supports inputs, selects, and textareas, to also support a file input as well.

I have some free time on my hands and would be happy to put the work in.

Sorry if this has already been proposed, I couldn't find anything about it in the issue tracker.

Thanks! 👋

**Comment by gustavohenke** [comment]
Duplicate of #1989

**Comment by seport** [comment]
@gustavohenke I don't think this is a duplicate. I'm not asking to add examples to the docs, I'm asking to add default file functionality to redux form. Am I misunderstanding that thread?

**Comment by gustavohenke** [comment]
One could want to store it as whatever a `FileReader` reads to -- what needs some async love to happen (but could be user-land) --, plus the `File` object itself.

**Comment by danielrob** [comment]
@seport Definitely encourage your research on this! If there's a sensible default, it'd be an nice feature ;).

**Comment by seport** [comment]
I'm up to code it, I just don't know what a sensible default would be that we could agree on. I'd propose saving the multipart file object, but it sounded like there was some contention around that.

**Comment by seport** [comment]
Since this thread has been up for a while, I'm thinking about giving it a first pass and getting a PR up as a proposal. Hopefully it could spawn some discussion?
